tests/e2e/cpu_specific/pack_i8_vnni.mlir

```diff
@@ -1,3 +1,5 @@
+// These i8 pack ops are seen in CPU codegen with AVX-512 Vector Neural Network Instructions (VNNI).
```

Comment:
this IR should be valid to run everywhere, right? this feels like more of a regression test that could be run on all targets?
(I'd assume cpu_specific would mean that it used ukernels/etc that only ran on CPU targets)

Reply:
(mostly just don't want to set the dangerous precedent that cpu devs will go dump generic tests in here because they work on cpu, or gpu in gpu_specific, etc - we want more tests in common than fewer)

Reply:
Chatted with @hanhanW about this test here: https://github.com/iree-org/iree/pull/17766/files#r1659260529

> I name it with `vnni` because of the inner tile sizes. They are the i8 pack ops we are seeing on CPU path (with vnni features). For other backends, they will likely see other pack ops, and they don't have to support vnni cases. It's a plus to handle it for functionality complement, but they don't really care about these pack ops.
> 
> The current status is: The vulkan-spirv backend does not support pack op codegen at this moment. I recently added the support for rocm side. CPU should be able to codegen all the pack ops.

Reply:
sounds like keeping it where it was and excluding it from spirv for now?
(users can author pack ops, and we need all backends to be able to handle them - may not be fast/optimal on them, but it should run)

Reply:
The IR does look general enough to me, yeah. I think the tile sizes uses with `tensor.pack` (https://mlir.llvm.org/docs/Dialects/TensorOps/#tensorpack-tensorpackop) are unhandled somewhere in non-CPU backends... errors were of the form
```
/work/tests/e2e/tensor_ops/pack_i8.mlir:30:11: error: 'func.func' op failed to get lowering configuration
  %pack = tensor.pack %source
          ^
```

Since this IR _could_ come from a frontend then maybe we should keep the test generic to all backends and just skip (ideally mark XFAIL) where unsupported today.

Reply:
Yeah, I _really_ don't like the idea of us ossifying in a state where tensor-level ops with valid parameters (that users/patterns/passes/etc can generate) cause explosions in backends - that's bad. Find if it's not supported yet, but we should be going for full support (even if slow) - otherwise autotuning/search/etc can't work effectively and users are going to be unhappy.